### DISCUSSION TITLE
LDM Notes for February 7th, 2024

### DISCUSSION THREAD

**Discussion by 333fred** [dotnet/csharplang#7938] [LDM Notes]
https://github.com/dotnet/csharplang/blob/main/meetings/2024/LDM-2024-02-07.md

### Agenda

- Partial type inference
- Breaking change warnings

With this, we are caught back up until the next time I fall behind again. Thanks for your patience while I got all the notes from the past few weeks out.

**Comment by yaakov-h** [comment]
What was the suggestion for changing `<LangVersion>Latest</LangVersion>`?

**Reply by jnm2** [reply]
From <https://github.com/dotnet/csharplang/issues/7918>:

> _**Alternative:**_ We could "park" the meaning of `latest` at C# 12, either with or without a warning to say that `latest` no longer works as you'd expect. This way, users would still get breaking change warnings, and would quickly be discouraged from using `latest`.

**Reply by KennethHoff** [reply]
It was a way to go around the fact that "latest" means the newest officially released that's supported by the SDK. If there renew l ends up being breaking changes to the language then simply installing a new SDK could potentially break your code - without even touching it!

**Reply by yaakov-h** [reply]
That's interesting. I've found `Latest` to be useful when multitargeting to older runtimes, but still use `global.json` so that installing a new SDK doesn't immediately upend the whole universe. Without that, yes it does sound quite problematic.

**Comment by jnm2** [comment]
> in case they did not migrate using whatever garden-path approach we create

I'm familiar with <https://en.wikipedia.org/wiki/Garden-path_sentence> (luring into an initial incorrect interpretation) and <https://www.merriam-webster.com/dictionary/lead%20%28someone%29%20down%2Fup%20the%20garden%20path> (deceiving), neither of which sound like a lovely approach to create 😄
